test/mitmproxy/tservers.py

```diff
@@ -122,6 +122,7 @@ def get_proxy_config(cls):
             cadir = cls.cadir,
             authenticator = cls.authenticator,
             add_upstream_certs_to_client_chain = cls.add_upstream_certs_to_client_chain,
+            ssl_insecure = True,
```

Comment:
It's fine if all tests are run using `--insecure`. We should add a tests that assures that we actually verify the certificate if `ssl_insecure` is not passed.